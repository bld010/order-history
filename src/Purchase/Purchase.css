.Purchase {
  width: 900px;
  margin: 50px;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  background-color: white;
}

.Purchase img {
  max-width: 90px;
}

.Purchase .center {
  width: 400px;
  display: flex;
  flex-direction: column;
  text-align: left;
  margin: 50px;
}

.Purchase .left {
  width: 120px;
}

.Purchase .center h3 {
  margin-bottom: 20px;
}

