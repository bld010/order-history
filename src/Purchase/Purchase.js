import React from 'react';
import './Purchase.css';

const Purchase = ({purchase, deletePurchase}) => {
  return(
    <article className="Purchase">
      <div className="left">
        <img src={purchase.img} alt={purchase.name}></img>
      </div>
      <div className="center">
        <h3>{purchase.name}</h3>
        <p>{purchase.description}</p>
      </div>
      <div className="right">
        <h3>${purchase.price}</h3>
        <button onClick={(d) => deletePurchase(e, purchase.id)}>Remove From History</button>
      </div>
    </article>
  )
} 

export default Purchase;