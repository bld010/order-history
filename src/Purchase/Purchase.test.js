import React from 'react';
import { shallow } from 'enzyme';
import Purchase from './Purchase';

describe('Purchase', () => {

  let mockPurchase = { 
      id: 1, 
      name: 'iPhone', 
      description: 'Expensive phone',
      img: 'www.apple.com/iphone.bmp',
      price: 900
     }

  let wrapper = null;
  let mockDeletePurchase = jest.fn()

    beforeEach(() => {

    wrapper = shallow(<Purchase 
      deletePurchase={mockDeletePurchase}
      purchase={mockPurchase}/>)
  })
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should fire deletePurchase when button is clicked', () => {
    
    
    wrapper.find('button').simulate('click')

    expect(mockDeletePurchase).toHaveBeenCalled();
  })
})