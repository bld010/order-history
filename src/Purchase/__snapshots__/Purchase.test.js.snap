// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Purchase should match the snapshot 1`] = `
<article
  className="Purchase"
>
  <div
    className="left"
  >
    <img
      alt="iPhone"
      src="www.apple.com/iphone.bmp"
    />
  </div>
  <div
    className="center"
  >
    <h3>
      iPhone
    </h3>
    <p>
      Expensive phone
    </p>
  </div>
  <div
    className="right"
  >
    <h3>
      $
      900
    </h3>
    <button
      onClick={[Function]}
    >
      Remove From History
    </button>
  </div>
</article>
`;
